xrootd/cmd/xrd-srv/main_test.go

```diff
@@ -204,3 +205,195 @@ func BenchmarkHandler_Dirlist(b *testing.B) {
 		}
 	}
 }
+
+func TestHandler_Open(t *testing.T) {
+	for _, tc := range []struct {
+		testName   string
+		file       string
+		mode       xrdfs.OpenMode
+		options    xrdfs.OpenOptions
+		createFile bool
+		errCode    xrdproto.ServerErrorCode
+	}{
+		{
+			testName:   "Readonly | created file",
+			options:    xrdfs.OpenOptionsOpenRead,
+			createFile: true,
+			file:       "test1.txt",
+		},
+		{
+			testName:   "Read & write | created file",
+			options:    xrdfs.OpenOptionsOpenUpdate,
+			createFile: true,
+			file:       "test1.txt",
+		},
+		{
+			testName:   "Append | created file",
+			options:    xrdfs.OpenOptionsOpenAppend,
+			createFile: true,
+			file:       "test1.txt",
+		},
+		{
+			testName: "Read & Write | new file",
+			options:  xrdfs.OpenOptionsOpenUpdate | xrdfs.OpenOptionsNew,
+			file:     "test1.txt",
+		},
+		{
+			testName:   "Read & Write | create existing file",
+			options:    xrdfs.OpenOptionsOpenUpdate | xrdfs.OpenOptionsNew,
+			createFile: true,
+			errCode:    xrdproto.IOError,
+			file:       "test1.txt",
+		},
+		{
+			testName:   "Read & Write | recreate file",
+			options:    xrdfs.OpenOptionsOpenUpdate | xrdfs.OpenOptionsDelete,
+			createFile: true,
+			file:       "test1.txt",
+		},
+		{
+			testName: "Read & Write | new file in new directory without OpenOptionsMkPath",
+			options:  xrdfs.OpenOptionsOpenUpdate | xrdfs.OpenOptionsNew,
+			file:     path.Join("testdir", "test1.txt"),
+			errCode:  xrdproto.IOError,
+		},
+		{
+			testName: "Read & Write | new file in new directory with OpenOptionsMkPath",
+			options:  xrdfs.OpenOptionsOpenUpdate | xrdfs.OpenOptionsNew | xrdfs.OpenOptionsMkPath,
+			file:     path.Join("testdir", "test1.txt"),
+			mode:     xrdfs.OpenModeOwnerRead | xrdfs.OpenModeOwnerWrite | xrdfs.OpenModeOwnerExecute,
+		},
+	} {
+		t.Run(tc.testName, func(t *testing.T) {
+			srv, addr, baseDir, err := createServer(func(err error) {
+				t.Error(err)
+			})
+			if err != nil {
+				t.Fatal(err)
+			}
+			defer srv.Shutdown(context.Background())
```

Comment:
in `createServer`, the `baseDir` is created first, then the `server`.
so... the order of the `defer`s should probably reversed?